src/storage-controller/src/lib.rs

```diff
@@ -746,7 +763,9 @@ where
             }
 
             // Assert some invariants.
-            if !dependency_read_holds.is_empty() {
+            //
+            // TODO(parkmycar): Include Tables (is_in_txns) in this check.
+            if !dependency_read_holds.is_empty() && !is_in_txns(id, &metadata) {
```

Comment:
Why is checking if the collection is in txn needed here?

Reply:
Expanded this comment!